proxy-node-shared/src/main/java/uk/gov/ida/notification/shared/proxy/ProxyNodeJsonClient.java

```diff
@@ -0,0 +1,35 @@
+package uk.gov.ida.notification.shared.proxy;
+
+import org.slf4j.MDC;
+import uk.gov.ida.jerseyclient.ErrorHandlingClient;
+import uk.gov.ida.jerseyclient.JsonClient;
+import uk.gov.ida.jerseyclient.JsonResponseProcessor;
+import uk.gov.ida.notification.shared.ProxyNodeMDCKey;
+
+import javax.inject.Inject;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ProxyNodeJsonClient {
+
+    private final JsonResponseProcessor responseProcessor;
+    private final ErrorHandlingClient errorHandlingClient;
+
+    @Inject
+    public ProxyNodeJsonClient(ErrorHandlingClient errorHandlingClient, JsonResponseProcessor responseProcessor) {
+        this.errorHandlingClient = errorHandlingClient;
+        this.responseProcessor = responseProcessor;
+    }
+
+    public <T> T post(Object postBody, URI uri, Class<T> clazz) {
+        return responseProcessor.getJsonEntity(uri, null, clazz, errorHandlingClient.post(uri, getJourneyIdHeader(), postBody));
```

Comment:
I think it would be better instead to make `ProxyNodeJsonClient` have in instance of `JsonClient` and proxy to its `post` method so we can abstract all of this away. We don't really want to have a custom implementation to make the HTTP call and handle the response